Haskell/syntax_test_haskell.hs

```diff
@@ -120,6 +136,8 @@
 
    instance TooMany Int where
 -- ^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.declaration.instance.haskell
+-- ^^^^^^^^ keyword.declaration..haskell
```

Comment:
`..`?

Should probably be `keyword.declaration.instance` or `keyword.definition.instance`. That second component should mirror whatever the C mode does with function declaration vs definition. I'm assuming that `class` is `keyword.declaration.class`?

Reply:
Good catch, I was probably debating which 3rd level scope to use there and got sidetracked, which explains the `..`. Relating Haskell operations onto other languages is annoying, because the `class` keyword seems more related to a trait and `instance` declares the implementation of such a trait for a type. This is confusing, especially seeing how `class` is very different from what it does in object-oriented languages.

We don't use `keyword.definition` for anything, afaik.

Reply:
> Relating Haskell declarations to other languages is annoying, because the class keyword declares something like a trait and instance declares the implementation of such a trait for a type. This is confusing, seeing how class is very different from what it does in object-oriented languages.

I kinda wanted to use the C/C++ precendent, since class vs instance *really* is very much like the declaration/definition split in that language. Unfortunately, neither mode actually detects declarations (probably because it wasn't really possible before branching), so we have nothing to go on.